jsonrpc/server_pkg_test.go

```diff
@@ -0,0 +1,312 @@
+package jsonrpc
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
```

Comment:
All of the following tests can be moved to `server_test.go`. These private functions can all be tested through the `HandleRequest()`. We should not be testing private functions unless we require a complicated setup. You are already defining handlers and incoming requests so I don't see why they cannot be moved to `server_test.go`.

Also, all these tests require names and should be run as sub-tests.

Reply:
Those were the tests I used while developing that functions, it will take some time to make sure we dont lose coverage on the edge cases while moving them.

Reply:
You can note the coverage percentage before moving and then make sure it is the same after moving them. Goland can easily help with that.

Reply:
I just ran it with jsonrpc package with  coverage and it sits at 94.4%

Reply:
done